src/gc-mmtk.h

```diff
@@ -0,0 +1,34 @@
+#ifdef MMTK_GC
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+JL_EXTENSION typedef struct _bigval_t {
+    size_t sz;
+#ifdef _P64 // Add padding so that the value is 64-byte aligned
+    // (8 pointers of 8 bytes each) - (2 other pointers in struct)
+    void *_padding[8 - 2];
+#else
+    // (16 pointers of 4 bytes each) - (2 other pointers in struct)
+    void *_padding[16 - 2];
+#endif
+    //struct jl_taggedvalue_t <>;
+    union {
+        uintptr_t header;
+        struct {
+            uintptr_t gc:2;
+        } bits;
+    };
+    // must be 64-byte aligned here, in 32 & 64 bit modes
+} bigval_t;
```

Comment:
MMTk doesn't use a couple of fields required by the Stock GC, such as the ones with respect to the linked list `next` and `prev`. 
That's why I added this here, and modified the `_padding` field instead to keep the alignment. Since the result is pretty much the same, and we're not saving any memory, I'm happy to have the struct being common to both GCs if you think that's better.

Reply:
If we can define `bigval_t` with a different layout for MMTk, then it is reasonable to keep it in the GC impl. But if `bigval_t` has to be the same layout as the original `bigval_t`, then it is basically a duplicate, and we should just use the original one.